src/api/creatures.py

```diff
@@ -188,29 +188,25 @@ def feed_creature(user_id: int, creature_id: int, treat_sku: str):
 
                 if treat_sku == stats["fav_treat"]:
                     gold_earned = 5
+                    change_in_happiness = min(10, remaining_happiness)
+                    change_in_affinity = min(5, remaining_affinity)
+                    message = f"{stats["name"]} devoured the treat!"
+
+                elif treat_sku == stats["hated_treat"]:
                     change_in_happiness = (
-                        10 if remaining_happiness >= 10 else remaining_happiness
+                        -5 if remaining_happiness <= 95 else (remaining_happiness - 100)
```

Comment:
1) this logic doesn't account for higher stages of evolution where max happiness > 100. for example, a creature with max happiness 500 and current happiness of 50 will have a `remaining_happiness = 450`. feeding this creature their hated treat would cause `change_in_happiness = 450 - 100 = 350`
2) i think maybe the way we defined remaining_happiness makes the subtraction too confusing to read in general. if `remaining_happiness = max_happiness - happiness`, then `remaining_happiness - max_happiness (100) = -happiness`. So maybe instead of this, write `max(-5, -1 * happiness)` instead?